lib/matplotlib/widgets.py

```diff
@@ -1967,15 +1967,12 @@ def set_animated(self, val):
 
     def closest(self, x, y):
         """Return index and pixel distance to closest index."""
-        pts = np.transpose((self.x, self.y))
+        pts = np.column_stack([self.x, self.y])
         # Transform data coordinates to pixel coordinates.
         pts = self.ax.transData.transform(pts)
-        diff = pts - ((x, y))
-        if diff.ndim == 2:
-            dist = np.sqrt(np.sum(diff ** 2, axis=1))
-            return np.argmin(dist), np.min(dist)
-        else:
-            return 0, np.sqrt(np.sum(diff ** 2))
+        diff = pts - [x, y]
+        dist = np.hypot(*diff.T)
+        return np.argmin(dist), np.min(dist)
```

Comment:
This could be slightly faster by not finding the minimum twice:
~~~
min_index = np.argmin(dist)
return min_index, dist[min_index]
~~~
Not quite sure if this is a relevant time-saver, but it seems this could be run with user interaction (don't know what ToolHandles actually does). So runtime could potentially be relevant.

Reply:
sure